### PR TITLE
Remove MetadataLoadContext dependency from AndroidAppBuilder

### PR DESCRIPTION
**Pull request by steveisok**
This will allow AndroidAppBuilder to be packaged without having unnecessary dependencies.

### CODE REVIEW

File: src/mono/netcore/sample/Android/AndroidSampleApp.csproj [hunk 1]
```diff
@@ -5,25 +5,28 @@
...
-             Targets="Build;Publish"/>
+    <Message Text="Packaged ID: %(RuntimePack.PackageDirectory)" Importance="high" />
   </Target>
 
   <UsingTask TaskName="AndroidAppBuilderTask" AssemblyFile="$(AndroidAppBuilderTasksAssemblyPath)"/>
 
-  <Target Name="BuildApp" DependsOnTargets="RebuildAndroidAppBuilder;Build">
+  <Target Name="BuildApp" AfterTargets="CopyFilesToPublishDirectory">
     <PropertyGroup>
       <AndroidAbi Condition="'$(Platform)'=='arm64'">arm64-v8a</AndroidAbi>
```
Comment (akoeplinger) [inline src/mono/netcore/sample/Android/AndroidSampleApp.csproj:31]: this shouldn't use `Platform`, but `RuntimeIdentifier` instead (GitHub won't allow me to comment on the whole block:

```diff
-      <AndroidAbi Condition="'$(Platform)'=='arm64'">arm64-v8a</AndroidAbi>
-      <AndroidAbi Condition="'$(Platform)'=='arm'">armeabi-v7a</AndroidAbi>
-      <AndroidAbi Condition="'$(Platform)'=='x64'">x86_64</AndroidAbi>
-      <AndroidAbi Condition="'$(AndroidAbi)'==''">$(Platform)</AndroidAbi>
+      <AndroidAbi Condition="'$(RuntimeIdentifier)'=='android-arm64'">arm64-v8a</AndroidAbi>
+      <AndroidAbi Condition="'$(RuntimeIdentifier)'=='android-arm'">armeabi-v7a</AndroidAbi>
+      <AndroidAbi Condition="'$(RuntimeIdentifier)'=='android-x86'">x86</AndroidAbi>
+      <AndroidAbi Condition="'$(RuntimeIdentifier)'=='android-x64' or '$(AndroidAbi)'==''">x86_64</AndroidAbi>
```

Or probably an even better fix would be to just pass `RuntimeIdentifier` into `AndroidAppBuilderTask` and do the conversion internally there.

**Review by akoeplinger — COMMENTED** [review]
A few suggestions, but looks good otherwise. It'll conflict with https://github.com/dotnet/runtime/pull/43333 so we'll see who gets in first ;)

File: src/mono/netcore/sample/Android/Makefile [hunk 2]
```diff
@@ -8,11 +8,11 @@ runtimepack:
 	../../../../.././build.sh Mono+Libs -os Android -arch $(MONO_ARCH) -c $(MONO_CONFIG)
 
 run:
-	$(DOTNET) build \
+	$(DOTNET) publish \
 	/p:TargetArchitecture=$(MONO_ARCH) \
 	/p:TargetOS=Android \
```
Comment (akoeplinger) [inline src/mono/netcore/sample/Android/Makefile:13]:
```suggestion
```

**Review by akoeplinger — APPROVED** [review]

**Comment by SamMonoRT** [comment]
@fanyang-mono  fyi - this may be needed for your PR as well. Slight conflict with #43333

**Pull request merged**